### PR TITLE
Feat 2.2.2 spatial hierarchy

### PR DESCRIPTION
**Pull request by maxmalynowsky**
Adds geometry hierarchy to the checks modules.

Misc changes:
- Filter checks using argparser or environmental variable to exclude checks that take a long time like overlaps and hierarchy.

Other:
- Branched from: https://github.com/mapaction/cod-ab-data-quality/pull/11

### CODE REVIEW

File: tests/checks/test_geometry_hierarchy.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from geopandas import GeoDataFrame
+
+from src.checks.geometry_hierarchy import main
+
+
+def test_mdg_geometry_hierarchy(iso3: str, gdfs: list[GeoDataFrame]) -> None:
```
Comment (Murray2015) [inline tests/checks/test_geometry_hierarchy.py:6]: I think this is likely to be a slow test due to the spatial join, right? How do you feel about making a Pytest mark for slow tests, such that they can be skipped if running tests for a feature that doesn't touch this? (I'll caveat that we should run all tests before committing, merging, etc. But I'm thinking of making the DX better during other feature development)

File: .env.example [hunk 2]
```diff
@@ -1,3 +1,7 @@
 # Comma separated list of ISO3 codes used by commands.
 # If empty, will run for all 249 ISO3 codes.
 ISO3=
+# Comma separated list of checks to include.
+CHECKS_INCLUDE=
```
[the remark this answers is not in the document]
Comment (maxmalynowsky) [reply .env.example:6]: Ah, good catch! What I had in mind is that all of these `.env` parameters should be optional, so if none of them are supplied, the pipeline processes everything. I added the following text:

```sh
# Comma separated list of checks to include.
# If empty, will include all checks.
CHECKS_INCLUDE=
# Comma separated list of checks to exclude.
# If empty, will exclude no checks.
CHECKS_EXCLUDE=
```

File: tests/checks/test_geometry_hierarchy.py (hunk 1 above)
Comment (maxmalynowsky) [reply tests/checks/test_geometry_hierarchy.py:8]: Another good catch! I added `@pytest.mark.slow` to all the geometry tests.

**Comment by maxmalynowsky** [comment]
Thanks for the feedback, I've added the suggestions you made.

**Comment by Murray2015** [comment]
All LGTM @maxmalynowsky, approved!

**Pull request merged**